### PR TITLE
Adding Port to resolver struct.

### PR DESCRIPTION
**Pull request by tristanmorgan**
Hi ,

This PR is an attempt to add the port to the resolvers struct. I use a domain specific resolver that runs on a non-standard port and I use a file like the following to tell my system how to reach it.

`/etc/resolver/10.in-addr.arpa`
```
nameserver 127.0.0.1
port 8600
timeout 5
```

from man resolver(5)

_Also, I understand this code is probably rubbish so if you decide to implement it another way I won't be offended._

### CODE REVIEW

File: dns/scutil/config.go [hunk 1]
```diff
@@ -81,6 +82,8 @@ func readMacOSDNS(ctx context.Context, getSCUtilDNS scutilExecutor) (Config, err
 			}
 		case strings.Contains(key, "options"):
 			currentResolver.MulticastDNS = strings.Contains(value, "mdns")
+		case strings.Contains(key, "port"):
+			currentResolver.Port = value
```
Comment (JohnStarich) [inline dns/scutil/config.go:86]: Let's store the port as an `int`, a bit like `InterfaceIndex` and `Order`.

Can you add an int parser here? Looks like the same logic from `order` could be reused here.

**Review by JohnStarich — CHANGES_REQUESTED** [review]
Thanks @tristanmorgan for the PR and test! Handling a port is good with me 👍

File: dns/scutil/config.go (hunk 1 above)
Comment (tristanmorgan) [reply dns/scutil/config.go]: I have updated my PR to use an int instead. if you have an elegant way to set the default port 53, please let me know.

**Review by JohnStarich — APPROVED** [review]

**Comment by JohnStarich** [comment]
Thanks @tristanmorgan! LGTM 👍

**Pull request merged**